tests/test_filter_algorithms.py

```diff
@@ -0,0 +1,386 @@
+#!/usr/bin/env python
+#
+# Copyright 2018 Pixar Animation Studios
+#
+# Licensed under the Apache License, Version 2.0 (the "Apache License")
+# with the following modification; you may not use this file except in
+# compliance with the Apache License and the following modification to it:
+# Section 6. Trademarks. is deleted and replaced with:
+#
+# 6. Trademarks. This License does not grant permission to use the trade
+#    names, trademarks, service marks, or product names of the Licensor
+#    and its affiliates, except as required to comply with Section 4(c) of
+#    the License and to reproduce the content of the NOTICE file.
+#
+# You may obtain a copy of the Apache License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the Apache License with the above modification is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied. See the Apache License for the specific
+# language governing permissions and limitations under the Apache License.
+#
+
+"""Test harness for the filter algorithms."""
+
+import unittest
+import copy
+
+import opentimelineio as otio
+
+
+class OTIOAssertions(object):
+    def assertJsonEqual(self, known, test_result):
+        """Convert to json and compare that (more readable)."""
+        self.maxDiff = None
+
+        known_str = otio.adapters.write_to_string(known, 'otio_json')
+        test_str = otio.adapters.write_to_string(test_result, 'otio_json')
+
+        self.assertMultiLineEqual(known_str, test_str)
```

Comment:
This class seems like it could be useful in a bunch of places.  It might more logically live in tests/utils.py, or maybe even its own file.

Reply:
Yes, good call.  I'll make an issue to bump this up, that might fit better in the next release than this one.